Framework/Kernel/inc/MantidKernel/DataService.h

```diff
@@ -84,11 +84,11 @@ template <typename T> class DLLExport DataService {
   public:
```

Comment:
I know this hasn't been changed here but there is a method `std::string objectName() const { return m_name; }` in `NamedObjectNotification` and it would be better if it returned `const std::string &` rather than just `std::string`. Could you make that change too?